modules/consentManagement.js

```diff
@@ -182,6 +189,101 @@ function lookupIabConsent(cmpSuccess, cmpError, hookConfig) {
   }
 }
 
+function lookupUspConsent(uspSuccess, cmpError, hookConfig) {
+  function handleCmpResponseCallbacks() {
+    const uspResponse = {};
+
+    function afterEach() {
+      if (uspResponse.usPrivacy) {
+        uspSuccess(uspResponse, hookConfig);
+      }
+    }
+
+    return {
+      consentDataCallback: function (consentResponse) {
+        uspResponse.usPrivacy = consentResponse.usPrivacy;
+        afterEach();
+      }
+    }
+  }
+
+  let callbackHandler = handleCmpResponseCallbacks();
+  let uspapiCallbacks = {};
+
+  // the following code also determines where the USP is located and uses the proper workflow to communicate with it:
+  // check to see if USP is found on the same window level as prebid and call it directly if so
+  // check to see if prebid is in a safeframe (with USP support)
+  // else assume prebid may be inside an iframe and use the IAB USP locator code to see if USP's located in a higher parent window. this works in cross domain iframes
+  // if the USP is not found, the iframe function will call the cmpError exit callback to abort the rest of the USP workflow
+  let f = window;
+  let uspFrame;
+  while (!uspFrame) {
+    try {
+      if (f.frames['__uspapiLocator']) uspFrame = f;
+    } catch (e) { }
+    if (f === window.top) break;
+    f = f.parent;
+  }
+
+  if (!uspFrame) {
+    return cmpError('USP not found.', hookConfig);
+  }
+
+  callUspWhileInIframe('getUSPData', uspFrame, callbackHandler.consentDataCallback);
+
+  if (!uspFrame) {
+    return cmpError('USP frame not found.', hookConfig);
+  }
+
+  callUspWhileInIframe('getUSPData', uspFrame, callbackHandler.consentDataCallback);
+
+  function callUspWhileInIframe(commandName, uspFrame, moduleCallback) {
+    /* Setup up a __uspapi function to do the postMessage and stash the callback. */
+    window.__uspapi = function (cmd, ver, callback) {
+      let callId = Math.random() + '';
+      let msg = {
+        __uspapiCall: {
+          command: cmd,
+          version: ver,
+          callId: callId
+        }
+      };
+      uspapiCallbacks[callId] = callback;
+      uspFrame.postMessage(msg, '*');
+    };
+
+    /** when we get the return message, call the stashed callback */
+    window.addEventListener('message', readPostMessageResponse, false);
+
+    // call uspapi
+    window.__uspapi(commandName, 1, uspIframeCallback);
+
+    function readPostMessageResponse(event) {
+      let res = typeof event.data === 'string' ? JSON.parse(event.data) : event.data;
+      if (res.__uspapiReturn && res.__uspapiReturn.callId) {
+        let i = res.__uspapiReturn;
+        if (typeof uspapiCallbacks[i.callId] !== 'undefined') {
+          uspapiCallbacks[i.callId](i.returnValue, i.success);
+          delete uspapiCallbacks[i.callId];
+        }
+      }
+    }
+
+    function removePostMessageListener() {
+      window.removeEventListener('message', readPostMessageResponse, false);
+    }
+
+    function uspIframeCallback(consentObject) {
```

Comment:
as mentioned in the comment for line 266, there are two arguments passed, but this is only handling one of them.

Reply:
depending on how you handle it, you may also want to pass the second argument to `moduleCallback()`